cnd/src/start_swap.rs

```diff
@@ -1,31 +1,41 @@
 use crate::{
-    db::{ForSwap, Save, Swap},
+    db::{ForSwap, Save},
     halight, hbit, herc20,
-    network::{comit::RemoteData, WhatAliceLearnedFromBob, WhatBobLearnedFromAlice},
-    protocol_spawner::{ProtocolSpawner, Spawn},
-    storage::{Load, Storage},
-    LocalSwapId, Protocol, Role, Side,
+    network::{RemoteData, WhatAliceLearnedFromBob, WhatBobLearnedFromAlice},
+    Load, LocalSwapId, Protocol, ProtocolSpawner, Role, Side, Spawn, Storage,
 };
-use chrono::offset::Utc;
+use chrono::NaiveDateTime;
 
-pub async fn start_swap(
+#[derive(Clone, Copy, Debug)]
+pub struct DecisionSwap {
```

Comment:
I don't understand the terminology used here? Why `Decision`?

Reply:
To my knowledge, it is because instances of this struct are used to make further decisions about swaps, i.e. which protocols and role it actually is :)

Reply:
These `DecisionSwap`s are only ever used to match and decide what information to load from the dabase.

Reply:
Because we purely use it for decision making. It only loads builds `meta` object to be able to reason about an actual `Swap. 

It was originally called `MetaSwap`, but we went for `Decision` instead because `Meta` is a very abstract term.

Note that I don't have a strong opinion about this. Do you @D4nte ? :)

Reply:
😄 we all answered at the same time - only noticed that now

Reply:
What they said :)

Reply:
Ok I see, we usually used the `meta` prefix which does not convey that much information but still better than `Decision` I think which is just plain confusing.

IMO, this is the metadata of the swap. It gives context to the content of the swap, the content being the asset amounts, token contract address, etc. Hence I think would be better to use `SwapMetadata` than `DecisionSwap`.

Reply:
Already used this joke today but here it is just for you @D4nte; I have some other data that relates to the `SwapMetadata`, I am going to call it `SwapMetaMetadata`.

Reply:
Isn't it `SwapMetadataMetadata`?

Reply:
Buffalo buffalo Buffalo buffalo buffalo buffalo Buffalo buffalo

Reply:
> It gives context to the content of the swap

What about `SwapContext`?

> Hence I think would be better to use SwapMetadata than DecisionSwap.

Since we are already bikeshedding: `data` doesn't add any value and `SwapMeta` sounds funny. By flipping it around you end up at `MetaSwap` again :man_shrugging: 

A few more ideas based on what you said @D4nte :

- `SwapSetting`: The _setting_ in which the swap takes place (Role + Protocols)
- `SwapEnvironment`
- `SwapComposition`: The elements a swap is composed of

Reply:
https://github.com/comit-network/comit-rs/pull/2796